docs/src/main/tut/add-support-for-complex-types.md

```diff
@@ -0,0 +1,246 @@
+the class above cannot be read automatically by PureConfig.
+
+#### Add support for Identifiable
+
+##### Method 1: Implement `ConfigReader[Identifiable]` manually
+
+Similarly to add support for simple types, it is possible to manually create a
+`ConfigReader[Identifiable]`:
+
+```tut:silent
+import com.typesafe.config._
+import pureconfig._
+import pureconfig.error._
+
+def extractId(obj: ConfigObject): Either[ConfigReaderFailures, String] =
+  Option(obj.get("id")) match {
+    case None =>
+      Left(ConfigReaderFailures(KeyNotFound("id", ConfigValueLocation(obj))))
+    case Some(id) =>
+      ConfigReader[String].from(id)
+  }
+
+def extractValue(obj: ConfigObject): Either[ConfigReaderFailures, Int] =
+  Option(obj.get("value")) match {
+    case None =>
+      Left(ConfigReaderFailures(KeyNotFound("value", ConfigValueLocation(obj))))
+    case Some(value) =>
+      ConfigReader[Int].from(value)
+  }
+
+implicit val identifiableConfigReader = ConfigReader.fromFunction({
+case obj: ConfigObject =>
+ Option(obj.get("type")) match {
+   case None =>
+     Left(ConfigReaderFailures(KeyNotFound("type", ConfigValueLocation(obj))))
+   case Some(t) if t.valueType() == ConfigValueType.STRING =>
+     t.unwrapped().asInstanceOf[String] match {
+       case "class1" => extractId(obj).right.map(new Class1(_))
+       case "class2" =>
+         for {
+           id <- extractId(obj).right
+           value <- extractValue(obj).right
+         } yield new Class2(id, value)
+       case x =>
+         Left(ConfigReaderFailures(CannotConvert(obj.toString,
+           "Identifiable", s"type has value $t instead of class1 or class2",
+           ConfigValueLocation(obj),
+           "")))
+     }
+   case Some(t) =>
+     Left(ConfigReaderFailures(CannotConvert(obj.toString,
+       "Identifiable", s"type has value $t instead of class1 or class2",
+       ConfigValueLocation(obj),
+       "")))
+ }
+case wrong =>
+  Left(ConfigReaderFailures(WrongType(wrong.valueType(), Set(ConfigValueType.OBJECT), ConfigValueLocation(wrong), "")))
+})
+```
+
+**pros**: The main feature of this method is that it take control over PureConfig on how to
+decode your configuration. PureConfig takes the manually defined `ConfigReader[Identifiable]`
+as blackbox and applies it when it has to read your configuration
+**cons**: The main feature of this method is also the drawback, because by taking control over 
+PureConfig in how to read sealed families of case classes, it prevents PureConfig from doing its 
+magic. For instance, it prevents PureConfig from using the `CorpoductHint` as strategy to read 
+`Identifiable`. This is, generally speaking, not a good idea. In theory you want PureConfig 
+features also for your custom class.
+
+##### Method 2: Use a Scala copy of the configuration
+
+PureConfig works out-of-the-box with Scala sealed familes of case classes and one simple solution
```

Comment:
Suggestion: I'd either break this into two sentences or reword it like "Because PureConfig ... classes, one simple ..."